AstroCal/Calendar.py

```diff
@@ -0,0 +1,85 @@
+from kivy.app import App
+from kivy.uix.button import Button
+from kivy.uix.boxlayout import BoxLayout
+from kivy.uix.gridlayout import GridLayout
+from kivy.uix.floatlayout import FloatLayout
+from kivy.uix.popup import Popup
+from datetime import datetime
+from datetime import time
+from functools import partial
+import swisseph as swe
+
+dates = datetime.now()
+firstday = dates.replace(day=1)
+startday = firstday.isoweekday()
+if startday == 7:
+    startday = 0
+def daysInMonth():
+    if dates.month in [1,3,5,7,8,10,12]:
+        return 31
+    elif dates.month in [4,6,9,11]:
+        return 30
+    elif dates.month == 2:
+        yr = dates.year
+        if yr % 4 == 0 and (yr % 100 != 0 or yr % 400 == 0):
+            return 29
+        else:
+            return 28
+def getRiseSet(year, month, day, celestial, status):
```

Comment:
getRiseSet is a function in the controller and a lot of this code should be in the core.py. I will have a look at this during my free time tomorrow and I am happy to make the necessary changes if you want me to.